### DISCUSSION TITLE
Ballistic_Force in detail

### DISCUSSION THREAD

**Discussion by Ongjming** [SmartlyDressedGames/Unturned-3.x-Community#3067] [Q&A] [answered]
While there are details on what Ballistic_Steps, Ballistic_Travel and Ballistic_Drop does, there isn't much information out there on Ballistic_Force...

(Just so I don't get it wrong, to my knowledge, Ballistic_Steps = Breaks in bullet travel time (Range) for Ballistic_Drop to be factored for the next Ballistic_Travel distance.
So Timberwolf with Range = 300, Ballistic_Steps = 24 and Ballistic_Travel =12.5 would meant that, for the Range of up to 300 meters, every 12.5 meters the bullet takes, there would be a default bullet drop of 0.002 meters/or some distance unit idk before it continues to the next 12.5 meters.)
(Hmm, what is the relationship between the Ballistic Steps+Travel and the time in seconds between your mouse click and your target getting affected by the hit? Or is it all instant hitscan for non projectiles?)

From what I could find out there, Ballistic_Force between 3000 and 2000 (The tank cannon vs the Rocket) meant a precision difference for the firer. I am assuming that the 3000 force meant that the projectile moves faster compared to 2000 force, hence hitting enemies faster before they shifted.

Is there any detail as to how or what is the default Ballistic_Drop applicable for Ballistic_Force values? (Would love more detail on what 0.002 on default Ballistic_Drop mean in actual distances/how it interacts in the calculations)
Default Projectile travel period is 30 seconds (Projectile_Lifespan). Since other Ballistics lines in .dat file are overridden when Ballistic_Force is present, the default Ballistic_Drop is applied linearly throughout the projectile's travel time?
Also what does the value in Ballistic_Force represents in the game aspect:
How it factors into calculation? Its unit is stated to be Newtons, does it interact with in game gravity setting as well?
What would be difference between say Ballistic_Force 50 and gun with range 50?
Does high gravity setting in game kinda render Ballistic_Force based weapon useless if gravity does affect it?
Projectile_Launch_Force_Multiplier exist, so what is the default launch force's formula with regards to Ballistic_Force values, and if it is 1, how far would a player be launched in normal gravity between Ballistic_Force of 50 and 5000 in terms of meters?

Thanks in advance for anyone who knows :)

**Comment by SDGNelson** [comment, marked as answer]
Sorry, this is poorly named, Ballistic_Force is directly used when adding launch force to the physics projectile like rocket launchers or tank missiles. It is not used by regular bullet ballistics. It is also multiplied by the magazine's Projectile_Launch_Force_Multiplier.

**Reply by Ongjming** [reply]
I do understand that it applies entirely only to guns with projectiles. Was just wondering if there was a layman explanation for the resulting effect of a projectile at Ballistic_Force 20 vs Ballistic_Force 1800, perhaps compared to a usual hitscan bullet or something.

Like maybe Ballistic_Force 20 would be kinda equivalent to a bullet with Range 2 and Ballistic_Drop 2 or something, except that it is traveling slightly slower + can arc back downwards from gravity.

(On rough testing)
Ballistic_Force 20 literally just dump the projectile around 1 meter around the avatar.
Ballistic_Force 1800 behaves closer to Rocket Launcher (Force 2000), could travel quite a distance albeit slightly slower + with more curves.

**Reply by SDGNelson** [reply]
Not sure if there is a good comparison because it depends on the mass of the projectile and is affected by the rigidbody drag, angular drag, level gravity, etc. I intend to rewrite throwables, projectiles, and bullets to all use the same deterministic physics at which point it will just specify a launch speed so it will be easier to understand then.
